uxarray/grid/grid.py

```diff
@@ -1749,6 +1751,29 @@ def get_kd_tree(
 
         return self._kd_tree
 
+    def get_spatial_hash(
+        self,
+        reconstruct: bool = False,
+    ):
+        """Get the SpatialHash data structure of this Grid that allows for
+        fast face search queries. Face searches are used to find the faces that
+        a list of points, in spherical coordinates, are contained within.
+
+        Parameters
+        ----------
+        reconstruct : bool, default=False
+            If true, reconstructs the spatial hash
+
+        Returns
+        -------
+        self._spatialhash : grid.Neighbors.SpatialHash
+            SpatialHash instance
+        """
```

Comment:
It would be nice to have a minimal example in the docstring here, showing how to query the data structure. Something like this adapted from the notebook could be good. 

```
Examples
--------
>>> import uxarray as ux
Open a grid from a file path
>>> uxgrid = ux.open_grid("grid_filename.nc")

Obtain SpatialHash instance
>>> spatial_hash = uxgrid.get_spatial_hash()

Query to find the face a point lies within in addition to its barycentric coordinates.
>>> face_ids, bcoords = spatial_hash.query([0.0, 0.0])
```

Reply:
Added this to the docstring in https://github.com/UXARRAY/uxarray/pull/1169/commits/0c0dd6021c04569d7a055450beba63c87d768000